sdcm/utils/decorators.py

```diff
@@ -333,3 +333,30 @@ def wrapper(*args, **kwargs):
                 raise UnsupportedNemesis("Capacity Issue") from ex
             raise
     return wrapper
+
+
+def optional_stage(stage_names: str | list[str]):
+    """
+    Decorator checks if the provided test stage is enabled in the test configuration
+    'optional_test_stages' parameter. If the stage is disabled, the decorated function is skipped.
+    More details can be found in https://github.com/scylladb/scylla-cluster-tests/blob/master/docs/faq.md
+
+    :param stage_names: str or list, name of the test stage(s)
+    """
+    stage_names = stage_names if isinstance(stage_names, list) else [stage_names]
+
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            # making import here, to work around circular import issue
+            from sdcm.cluster import TestConfig
+            optional_test_stages = TestConfig().tester_obj().optional_test_stages
+            skipped_stages = [stage for stage in stage_names if not optional_test_stages[stage]]
```

Comment:
this logic slightly differs from `def optional_stage_enabled` and seem to raise exception if a stage is not in `optional_test_stages`. Shouldn't reuse this function here?

Reply:
in tester.py the defaultdict is used to define `optional_test_stages` attr, where it returns default value for keys not present in the dict.
So both version will work, but i prefer this one, will update the function to use `optional_test_stages[stage]` as well.

Reply:
hmm, I see, but this defaultdict returns `True` as default, shouldn't be False?

Reply:
maybe we should rename "SCT_OPTIONAL_TEST_STAGES" to "SCT_ENABLED_TEST_STAGES"?

Reply:
For me the `optional` version seems to be more descriptive - the name suggests that there could be optional stages in an SCT, and value defines which of them are (in the particular run).
WDYT?

Reply:
the name optional fit the description of the feature

but @soyacz is correct that it not completely clear as to what setting something to true in that mapping.

true might be that it's optional, while the the true is that is enabled (mandatory to run)

so @soyacz suggestion doesn't clarify the situation a bit

Reply:
My approach to naming the parameter was:
- the parameter name implies that there is a feature of optionality of test stages. And when there is a value set for the parameter (in contrast to empty default value) then the feature is enabled
- the parameter value is mapping of such optional test stages names to an indication whether they will be executed during test run or not

Maybe we can combine suggestions and use something like `enabled_optional_test_stages`?

Reply:
from my side, word 'optional' means 'not run' by default - like when buying a car, you don't start with all the optional features on by default. And for typical run (not reuse cluster) these are not optional usually.
This is why I'm bit confused with it. Perhaps I'm the only one.

Reply:
another approach would be by inverting logic, and make "skip_test_stages"